roofit/hs3/src/JSONFactories_RooFitCore.cxx

```diff
@@ -545,25 +545,25 @@ class RooFormulaArgStreamer : public RooFit::JSONIO::Exporter {
       const RooArg_t *pdf = static_cast<const RooArg_t *>(func);
       elem["type"] << key();
       TString expression(pdf->expression());
-      std::vector<std::pair<RooAbsArg *, std::size_t>> paramsWithIndex;
+      std::vector<std::pair<std::size_t, const RooAbsArg*>> paramsWithIndex;
```

Comment:
```suggestion
      std::vector<const RooAbsArg*> paramsWithIndex;
```
As you suggested it now, the vector is now sorted by the index (see one of the following lines):
```txt
std::sort(paramsWithIndex.begin(), paramsWithIndex.end());
```
However, in this case the information of the index is redundant. The sort does nothing at all. The original code was badly designed.

Could you please refactor the code such that this redundant index is not stored and there is no `std::sort`? That means that the loops will then look something like this:
```c++
      for (std::size_t i = 0; i < params.size(); ++i) {
          RooAbsArg const *par = params[idx];
	  expression.ReplaceAll(("x[" + std::to_string(idx) + "]").c_str(), par->GetName());
      }
```

Reply:
I realised even the vector is not really needed, so everything is now simplified to run in one (reverse order) loop.